products/sle/main.pm

```diff
@@ -418,6 +419,47 @@ logcurrentenv(
       SLE_PRODUCT SPLITUSR VIDEOMODE)
 );
 
+sub parse_condition {
+    my ($schedule, $condition_name) = @_;
+    # Get condition hash
+    my $condition = $schedule->{conditional_schedule}->{$condition_name};
+    # Iterate over variables in the condition
+    foreach my $var (keys %{$condition}) {
+        next unless my $val = get_var($var);
+        # If value of the variable matched the conditions
+        # Iterate over the list of the modules to be loaded
+        for my $test (@{$condition->{$var}->{$val}}) {
+            loadtest $test;
+        }
+    }
+}
+
+sub parse_test_suite_setting {
+    my ($setting) = shift;
+    # allow comma spearated list
+    for my $test (split(',', get_var($setting))) {
+        loadtest $test;
+    }
+}
+
+if (my $yamlfile = get_var('YAML_SCHEDULE')) {
+    my $schedule = YAML::Tiny::LoadFile(dirname(__FILE__) . '/../../' . $yamlfile);
+    #schedule/lvm_thin_provisioning.yml
+    for my $module (@{$schedule->{schedule}}) {
+        if ($module =~ s/\{\{(.*)\}\}/$1/) {
+            # load modules by evaluating conditions
+            parse_condition($schedule, $module);
```

Comment:
It is a little deceiving the names `parse_condition` or `parse_test_suite_setting` as they are doing some processing and loading modules, it would be more clear if both would return list of modules, so we just need to iterate them in #458 (initializing the list to $module). But it is just a matter of taste, no need to change it, just for separating the processing from the loading.